File: desafio.js
// criando a classe principal
class genericHero {
    constructor(name, age, heroType) {
        this.name = name
        this.age = age
        this.heroType = heroType
    }

    atack(){
        if (this.heroType === "mago"){
            this.atack = "magia"
            console.log(`O ${this.heroType} atacou usando ${this.atack}`)
        } 

        else if (this.heroType === "guerreiro"){
            this.atack = "espada"
            console.log(`O ${this.heroType} atacou usando ${this.atack}`)
        } 

        else if (this.heroType === "monge"){
            this.atack = "artes marciais"
            console.log(`O ${this.heroType} atacou usando ${this.atack}`)
        } 

        else if (this.heroType === "ninja"){
            this.atack = "shuriken"
            console.log(`O ${this.heroType} atacou usando ${this.atack}`)
        }

        else {
            console.log("Tipo de herói inválido. Insira um valor dentre as opções dadas.")
        }
    }
}

// Mensagem inicial
console.log("----------------------")
console.log("Bem vindo a sua aventura. Você tem os seguintes tipos de heróis: 'mago', 'guerreiro', 'monge' e 'ninja'.")
console.log("----------------------")

// Criar um objeto na classe
heroAtack = new genericHero("generic_name", 30, "mago")

// Chamar o método atack
heroAtack.atack()

// Mensagem final 
if (heroAtack.heroType === "mago" || heroAtack.heroType === "guerreiro" || heroAtack.heroType === "monge" || heroAtack.heroType === "ninja"){
    console.log("----------------------")
    console.log(`O herói ${heroAtack.name} está avançando em sua aventura de ${heroAtack.heroType}!`)
}

File: desafio.py
# Define a classe principal
class GenericHero:
    def __init__(self, name, age, hero_type):
        self.name = name
        self.age = age
        self.hero_type = hero_type

    def atack(self):
        if self.hero_type == "mago":
            atack = "magia"
            print(f"O {self.hero_type} atacou usando {atack}.")
        
        elif self.hero_type == "guerreiro":
            atack = "espada"
            print(f"O {self.hero_type} atacou usando {atack}.")
        
        elif self.hero_type == "monge":
            atack = "artes marciais"
            print(f"O {self.hero_type} atacou usando {atack}.")
        
        elif self.hero_type == "ninja":
            atack = "shuriken"
            print(f"O {self.hero_type} atacou usando {atack}.")
        
        else:
            print("Tipo de herói desconhecido, não pode atacar!")

# Mensagem inicial
print("----------------------")

print("Bem vindo a sua aventura. Você pode escolher entre os seguintes tipos de heróis: 'mago', 'guerreiro', 'monge' e 'ninja'. Escolha com cuidado! ")

print("----------------------")

# Criar um objeto da classe
hero_atack = GenericHero(input("digite o nome do seu personagem: "), input("digite a idade do seu personagem: "), input("digite o tipo do seu personagem: "))

# Chamar o método atack
hero_atack.atack()
print("----------------------")

# Mensagem final 
print(f"O herói {hero_atack.name} está avançando em sua aventura de {hero_atack.hero_type}!")